src/status.rs

```diff
@@ -513,35 +518,114 @@ struct FilteredTx<T> {
     result: Vec<T>,
 }
 
-fn filter_block_txs<T: Send>(
-    block: Block,
-    map_fn: impl Fn(&Transaction) -> Vec<T> + Sync,
-) -> impl Iterator<Item = FilteredTx<T>> {
-    block
-        .txdata
-        .into_par_iter()
-        .enumerate()
-        .filter_map(|(pos, tx)| {
-            let result = map_fn(&tx);
-            if result.is_empty() {
-                return None; // skip irrelevant transaction
+fn filter_block_txs_outputs(
+    block: SerBlock,
```

Comment:
this function and `filter_block_txs_inputs` below do not benefit from taking ownership of block (only used in Block::visit), this can as well be `&[u8]`

Reply:
thanks, fixed in e80a96365e0b6cd3bc461359327bd9c1fb729495